src/sundials/fmod/CMakeLists.txt

```diff
@@ -15,32 +15,10 @@
 # ---------------------------------------------------------------
 
 set(sundials_SOURCES
-  fsundials_adaptcontroller_mod.c
-  fsundials_adaptcontroller_mod.f90
-  fsundials_context_mod.c
-  fsundials_context_mod.f90
-  fsundials_futils_mod.c
-  fsundials_futils_mod.f90
-  fsundials_linearsolver_mod.c
-  fsundials_linearsolver_mod.f90
-  fsundials_logger_mod.c
-  fsundials_logger_mod.f90
-  fsundials_matrix_mod.c
-  fsundials_matrix_mod.f90
-  fsundials_nonlinearsolver_mod.c
-  fsundials_nonlinearsolver_mod.f90
-  fsundials_nvector_mod.c
-  fsundials_nvector_mod.f90
-  fsundials_types_mod.c
-  fsundials_types_mod.f90
+  fsundials_core_mod.c
+  fsundials_core_mod.f90
   )
 
-if(SUNDIALS_BUILD_WITH_PROFILING)
-  list(APPEND sundials_SOURCES
-    fsundials_profiler_mod.f90
-    fsundials_profiler_mod.c)
-endif()
```

Comment:
When profiling is disabled, `sundials_profiler.c` is not part of the `sundials_core` library, does the Fortran interface need to exclude `%include "fsundials_profiler.i"`?

Reply:
I think we should actually just compile in `sundials_profiler.c` even when profiling is disabled. It will have no effect other than the `SUNProfiler` functions could still be called by a user. Previously, we did not compile it in since it required C99 and we were targeting C89.